golos.ctrl/src/golos.ctrl.cpp

```diff
@@ -279,12 +282,20 @@ void control::update_witnesses_weights(vector<name> witnesses, share_type diff)
             print("apply_vote_weight: witness not found\n");
         }
     }
+
     update_auths();
 }
 
 void control::update_auths() {
     // TODO: change only if top changed #35
     auto top = top_witnesses();
+    auth_witnesses_tl top_witnesses(_self, _self.value);
+    const auto &old_top = top_witnesses.get().witnesses;
+    if ( std::equal(old_top.begin(), old_top.end(), top.begin()) ) {
```

Comment:
Will this detect that [acc1,acc2,acc3] is the same as [acc1,acc3,acc2] ?
Propose sort `top` by name. There is no need to sort `old_top`, it will be already sorted when you store current `top`